interpreter/decorators.go

```diff
@@ -42,6 +42,11 @@ func decObserveEval(observer EvalObserver) InterpretableDecorator {
 				InterpretableConst: inst,
 				observer:           observer,
 			}, nil
+		case InterpretableConstructor:
+			return &evalWatchConstructor{
```

Comment:
Don't forget to add `*evalWatchConstructor` to the list of early returns at the top of this switch statement.

Reply:
Right, fixed